`SuperSelect` web docs (HDS-3222) (#2026)
hashicorp/design-system: website/docs/components/form/super-select/partials/specifications/states.md

Reviewer: Updated this (current context added)

## States

!!! Info

**A note on disabled states**

> Because disabled states completely remove the interactive function of an element, it can be challenging for a user to understand why it has been disabled and/or why they cannot interact with that element. In an effort to avoid this confusion, we opt for using methods like enabling or hiding the element and, thus, are not offering disabled states for the Dropdown. Read more about [when to enable vs hide](https://hashicorp.atlassian.net/wiki/spaces/DES/pages/2678685874/Hiding+Disabling).

Reviewer 2: ```suggestion
Because disabled states completely remove the interactive function of an element, it can be challenging for a user to understand why it has been disabled and/or why they cannot interact with that element. In an effort to avoid this confusion, we opt for using methods like displaying or hiding the element and, thus, are not offering disabled states for the Dropdown. Read more about [when to enable vs hide](https://hashicorp.atlassian.net/wiki/spaces/DES/pages/2678685874/Hiding+Disabling).
```

> Because disabled states completely remove the interactive function of an element, it can be challenging for a user to understand why it has been disabled and/or why they cannot interact with that element. In an effort to avoid this confusion, we opt for using methods like displaying or hiding the element and, thus, are not offering disabled states for the Dropdown. Read more about [when to enable vs hide](https://hashicorp.atlassian.net/wiki/spaces/DES/pages/2678685874/Hiding+Disabling).

Reviewer: If I'm not mistaken, we are talking about the list for the SuperSelect. When the SuperSelect is disabled, we do not allow a disabled list to be shown (because you cannot open it in that state thus it's not available). So we should probably reword this to say "SuperSelect list" vs "dropdown."

```suggestion
Because disabled states completely remove the interactive function of an element, it can be challenging for a user to understand why it has been disabled and/or why they cannot interact with that element. In an effort to avoid this confusion, we opt for using methods like displaying or hiding the element and, thus, are not offering disabled states for the SuperSelect list. Read more about [when to enable vs hide](https://hashicorp.atlassian.net/wiki/spaces/DES/pages/2678685874/Hiding+Disabling).
```

> Because disabled states completely remove the interactive function of an element, it can be challenging for a user to understand why it has been disabled and/or why they cannot interact with that element. In an effort to avoid this confusion, we opt for using methods like displaying or hiding the element and, thus, are not offering disabled states for the Dropdown. Read more about [when to enable vs hide](https://hashicorp.atlassian.net/wiki/spaces/DES/pages/2678685874/Hiding+Disabling).

Author: @majedelass I think the wording is still a bit confusing.

The SuperSelect as a whole can be disabled which makes the "trigger" non-interactive so when you click on it nothing happens (ie the dropdown with the list of options will not display.)

Individual option items can also be disabled. This means that when you click on them within the dropdown list nothing happens. It is still possible to set the disabled options to be pre-selected however. (Alex just updated the Showcase to include examples of these and I fixed some related styles.)

> Because disabled states completely remove the interactive function of an element, it can be challenging for a user to understand why it has been disabled and/or why they cannot interact with that element. In an effort to avoid this confusion, we opt for using methods like displaying or hiding the element and, thus, are not offering disabled states for the Dropdown. Read more about [when to enable vs hide](https://hashicorp.atlassian.net/wiki/spaces/DES/pages/2678685874/Hiding+Disabling).

Reviewer: Maybe we should reword this to just say: 

```suggestion
Because disabled states completely remove the interactive function of an element, it can be challenging for a user to understand why it has been disabled and/or why they cannot interact with that element. In an effort to avoid this confusion, we opt for using methods like displaying or hiding the element. Read more about [when to enable vs hide](https://hashicorp.atlassian.net/wiki/spaces/DES/pages/2678685874/Hiding+Disabling).
```

> Because disabled states completely remove the interactive function of an element, it can be challenging for a user to understand why it has been disabled and/or why they cannot interact with that element. In an effort to avoid this confusion, we opt for using methods like displaying or hiding the element and, thus, are not offering disabled states for the Dropdown. Read more about [when to enable vs hide](https://hashicorp.atlassian.net/wiki/spaces/DES/pages/2678685874/Hiding+Disabling).

Author: Thanks @majedelass this looks better. Applying...

!!!

### Trigger

![Showing the different states for the SuperSelect trigger](/assets/components/form/super-select/super-select-trigger-states.png =800x*)

### ListItem

#### Checkmark

> ![Showing the different states for the SuperSelect list item checkmark variant](/assets/components/form/super-select/super-select-listitem-checkmark-states.png =432x*)

Reviewer 3: Looking at the CSS, I _think_ the disabled version of the checkmark doesn't look like this. @KristinLBradley / @alex-ju can you double check?

Author: I found a reference in the docs on setting disabled options: https://ember-power-select.com/docs/the-list (Originally I was looking in the API docs and not finding anything) I will add an example to the Showcase.

Author: Added to the Showcase in the component branch

Author: When an option is disabled it can't be manually selected. However, it seems that it can be preselected using `@selected`.

We don't have a special visual style for this. Should it perhaps be both "checked" and use a grey text color (vs. blue)?

Author: @alex-ju pushed up a number of updates related to this. I also pushed up some fixes to styles that hadn't been working properly.
